src/lib/util/collapseHourlyMetrics.ts

```diff
@@ -0,0 +1,41 @@
+import { IClientMetricsEnv } from '../types/stores/client-metrics-store-v2';
+import { startOfHour } from 'date-fns';
+
+const sum = (items: number[]): number => {
+    return items.reduce((acc, item) => acc + item, 0);
+};
+
+const groupBy = <T>(list: T[], createKey: (item: T) => string): T[][] => {
+    const groups = list.reduce((acc, item) => {
+        const key = createKey(item);
+        acc[key] = acc[key] ?? [];
+        acc[key].push(item);
+        return acc;
+    }, {} as Record<string, T[]>);
+
+    return Object.values(groups);
+};
+
+const createMetricKey = (metric: IClientMetricsEnv): string => {
+    return [
+        metric.featureName,
+        metric.appName,
+        metric.environment,
+        metric.timestamp.getTime(),
+    ].join();
+};
+
+export const collapseHourlyMetrics = (
+    metrics: IClientMetricsEnv[],
+): IClientMetricsEnv[] => {
+    const hourlyMetrics = metrics.map((metric) => ({
+        ...metric,
+        timestamp: startOfHour(metric.timestamp),
```

Comment:
Here I believe you might be overriding the `timestamp` of the original metric. If it's intended that's fine, otherwise, maybe you can move this to the group by function sou you'd be using `startOfHour` in the group by but maintaining the original timestamp of the metric

Reply:
Yep intended, though a bit unclear. The function is supposed to group metrics by start-of-hour (like in the original code).